pkg/manager/manager_test.go

```diff
@@ -0,0 +1,110 @@
+package manager
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"gopkg.in/fsnotify.v1"
+
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context/fake"
+)
+
+func TestManager_FileWatch(t *testing.T) {
```

Comment:
Since the test code has a lotta if/else logic to test the failure scenarios, what might make the test code more readable is to have 3 separate `t.Run()` calls with each of the scenarios.